pkg/utils/utils.go

```diff
@@ -258,3 +292,124 @@ func CleanCachedNetConf(cRefPath string) error {
 	}
 	return nil
 }
+
+// htons converts an uint16 from host to network byte order.
+func htons(i uint16) uint16 {
+	return (i<<8)&0xff00 | i>>8
+}
+
+// SendGratuitousArp sends a gratuitous ARP packet with the provided source IP over the provided interface.
+func SendGratuitousArp(srcIP net.IP, iface net.Interface) error {
```

Comment:
I still need to give this PR a deeper look.


i see : https://github.com/mdlayher/arp implements sending arp.
perhaps we can use here ? 


in any case i feel we should put the logic which builds the packets  (for both arp and and NA) in a separate file 

e.g `utils/packet.go`

Reply:
I took a look at the implementation of the library. I don't believe it supports sending gratuitous ARPs. Rather it is a library for ARP resolution. This library supports gratuitous ARPs: https://github.com/j-keck/arping . However I am not in favor of it's implementation since it uses raw sockets and builds the entire Ethernet Frame. (https://github.com/j-keck/arping/blob/c50f4319d307d95d1e2b344c4504158705e919de/arping_linux.go#L21)

Reply:
I looked at the arping library that you mentioned. At the API level, it doesn't seem to expose that it builds the full packet including ethernet. I'm in favor of still relying on that external library to build the packet, or patching that library to (add an option to) not use raw sockets.

In any case, it feels like we are just dumping "a small arp library" into `pkg/utils/utils.go` where we are already doing all kinds of other things.

Reply:
I put all the logic for sending packets (ARPs/Unsolicited NA) in a separate file. I hope this works for everyone.